src/components/Upload/index.jsx

```diff
@@ -110,15 +110,18 @@ class Upload extends React.Component {
         // Once upload is done, create a resource
         const ckanResource = frictionlessCkanMapper.resourceFrictionlessToCkan(
           resource.descriptor
-        );
-        delete ckanResource.sample;
-        client.action(
-          "resource_create",
-          Object.assign(ckanResource, {
-            package_id: this.state.datasetId,
+          );
+          delete ckanResource.sample;
+          client.action(
+            "resource_create",
+            Object.assign(ckanResource, {
+              package_id: this.state.datasetId,
+              name: resource.descriptor.name
+            })
+            ).then(response => {
+              this.props.onChangeResourceId(response.result.id)
+            })
```

Comment:
Where is the error handling if this goes wrong in some way?

Reply:
I'm not sure, but the catch below should set `error` as `true`. But I need to reproduce it to be sure.
```
.catch((error) => {
        this.setState({ error: true, loading: false });
        this.props.handleUploadStatus({
          loading: false,
          success: false,
          error: true,
        });
      });
```

But definitely should be good add a new catch only for this request and create a new error message too.